[DoctrineMesseger] Incompatibility with dbal 4 on Mysql

### Symfony version(s) affected

7.3 | 7.4-BETA | 7.4-RC

### Description

Doctrine messenger uses the type `DATETIME_IMMUTABLE` for the columns `created_at`, `available_at`, `delivered_at` on table `messenger_messages`.

From `Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection.php` in method `addTableToSchema`
```php
        $table->addColumn('created_at', Types::DATETIME_IMMUTABLE)
            ->setNotnull(true);
        $table->addColumn('available_at', Types::DATETIME_IMMUTABLE)
            ->setNotnull(true);
        $table->addColumn('delivered_at', Types::DATETIME_IMMUTABLE)
```

This leads to the creation of the columns with a comment `DC2Type` to map back

<img width="1428" height="299" alt="Image" src="https://github.com/user-attachments/assets/267a2e4a-ddef-423b-bae1-25e6e7661e8e" />



After migrating to DBAL 4+, the comments are not possible anymore and are ignored, with the consequence of the command `php bin/console make:migration` creating this migration:

```php
    public function up(Schema $schema): void
    {
        // this up() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE messenger_messages CHANGE created_at created_at DATETIME NOT NULL, CHANGE available_at available_at DATETIME NOT NULL, CHANGE delivered_at delivered_at DATETIME DEFAULT NULL');
    }

    public function down(Schema $schema): void
    {
        // this down() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE messenger_messages CHANGE created_at created_at TIMESTAMP NOT NULL COMMENT \'(DC2Type:datetime_immutable)\', CHANGE available_at available_at TIMESTAMP NOT NULL COMMENT \'(DC2Type:datetime_immutable)\', CHANGE delivered_at delivered_at TIMESTAMP DEFAULT NULL COMMENT \'(DC2Type:datetime_immutable)\'');
    }
```

The first interesting thing is that in the `down` method there is  `TIMESTAMP`, but my column is of type `DATETIME`.

The second problem is that executing the same command after will then create this over and over:

```php
    public function up(Schema $schema): void
    {
        // this up() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE messenger_messages CHANGE created_at created_at DATETIME NOT NULL, CHANGE available_at available_at DATETIME NOT NULL, CHANGE delivered_at delivered_at DATETIME DEFAULT NULL');
    }

    public function down(Schema $schema): void
    {
        // this down() migration is auto-generated, please modify it to your needs
        $this->addSql('ALTER TABLE messenger_messages CHANGE created_at created_at TIMESTAMP NOT NULL, CHANGE available_at available_at TIMESTAMP NOT NULL, CHANGE delivered_at delivered_at TIMESTAMP DEFAULT NULL');
    }
```

In the `down` method there is still  `TIMESTAMP`, while my column is of type `DATETIME`

### How to reproduce

Just install the package with mysql database and DBAL 3. Then migrate to DBAL 4

### Possible Solution

As a workaround, in my `doctrine.yaml`, I put the following: `doctrine.dbal.schema_filter: ~^(?!messenger_messages)~`

It will work for the migration:
```
 > php bin/console make:migration

[WARNING] No database changes were detected.
 The database schema and the application mapping information are already in sync
```

But won't work for other commands like:

```
 > php bin/console doctrine:schema:validate

Mapping
-------
[OK] The mapping files are correct.

Database
--------
[ERROR] The database schema is not in sync with the current mapping file. 
```

```
 >  php bin/console doctrine:schema:update --dump-sql
ALTER TABLE messenger_messages CHANGE created_at created_at DATETIME NOT NULL, CHANGE available_at available_at DATETIME NOT NULL, CHANGE delivered_at delivered_at DATETIME DEFAULT NULL;
```

### Additional Context

_No response_

--- a/src/Symfony/Component/Messenger/Bridge/Doctrine/Tests/Transport/ConnectionTest.php
+++ b/src/Symfony/Component/Messenger/Bridge/Doctrine/Tests/Transport/ConnectionTest.php
@@ -32,6 +32,7 @@ use Doctrine\DBAL\Schema\AbstractSchemaManager;
 use Doctrine\DBAL\Schema\NamedObject;
 use Doctrine\DBAL\Schema\Schema;
 use Doctrine\DBAL\Schema\SchemaConfig;
+use Doctrine\DBAL\Types\DateTimeType;
 use PHPUnit\Framework\Attributes\DataProvider;
 use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
@@ -815,6 +816,10 @@ class ConnectionTest extends TestCase
             }
         }
         $this->assertTrue($hasCoveringIndex, 'Expected covering index on [queue_name, available_at, delivered_at, id] not found');
+
+        $this->assertInstanceOf(DateTimeType::class, $table->getColumn('created_at')->getType());
+        $this->assertInstanceOf(DateTimeType::class, $table->getColumn('available_at')->getType());
+        $this->assertInstanceOf(DateTimeType::class, $table->getColumn('delivered_at')->getType());
     }
 
     public function testConfigureSchemaDifferentDbalConnection()

--- a/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php
+++ b/src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php
@@ -518,11 +518,12 @@ class Connection implements ResetInterface
         $table->addColumn('queue_name', Types::STRING)
             ->setLength(190) // MySQL 5.6 only supports 191 characters on an indexed column in utf8mb4 mode
             ->setNotnull(true);
-        $table->addColumn('created_at', Types::DATETIME_IMMUTABLE)
+        // Keep schema types as mutable datetime to avoid perpetual MySQL diffs with DBAL 4+.
+        $table->addColumn('created_at', Types::DATETIME_MUTABLE)
             ->setNotnull(true);
-        $table->addColumn('available_at', Types::DATETIME_IMMUTABLE)
+        $table->addColumn('available_at', Types::DATETIME_MUTABLE)
             ->setNotnull(true);
-        $table->addColumn('delivered_at', Types::DATETIME_IMMUTABLE)
+        $table->addColumn('delivered_at', Types::DATETIME_MUTABLE)
             ->setNotnull(false);
         if (class_exists(PrimaryKeyConstraint::class)) {
             $table->addPrimaryKeyConstraint(new PrimaryKeyConstraint(null, [new UnqualifiedName(Identifier::unquoted('id'))], true));

